### ISSUE TITLE
[BUG]update feed action doesnt get new posts for a feed that had errors just before

### ISSUE THREAD

**Issue by rezad1393** [FreshRSS/FreshRSS#4477]
**Describe the bug**
when a feed has a error and can be updated ( or a new feed that user is trying to add) then somehow freshrss keeps the feed status and wont get new post even after the error is gone.

**To Reproduce**
this is hard to reproduce but I will try.
there are two situation.
1) 
add a feed that you cant temporary access.
after sometime trying to add it (10 seconds?) the freshrss goes back to empty page of "add new feed" and says to check log.
if you try again very soon, like after first try, then when you again try to add the feed, it seems that frehsrss app doesn't check the current status of feed and just uses the old error status and wont let you add the feed.

2)when a feed that you added before and you try to manually get new post by using update feed button has a error, then that feed wont get updated, and freshrss main page is shown with all feeds post ,Which is an indication that feed update failed (but users doesn't know that  so many that is a feature request too , to show the user the error but main page).
if I try to again update the feed again after that error (which doesn't has the error anymore) , the page of the feed is shown with no new post even though I can use direct rss url and see the new posts in browser.
I literally see this right now that I refresh a feed (With the update feeds button) and first time after some time it went to main page, so it had a error ( meaning my network was slow and feed timed out ).
but when I tries again the button , then it goes to the feed page in freshrss UI, so it thinks it worked and got all the feed.
even when I use the button 'delete all articles' and then update the feeds button, it get the post up to the last time it correctly got them and doesnt get the post that are on rss url posted after that error happened.

maybe frehrss uses a time delay between trying to update feeds that have errors and before that just uses cache?
can this be changed?
cache for successful update is a good idea but caching failed feed (updating cache time for failed feeds ) is a bad idea that cause this issue.

feel free to ask me if my issue was not clear.

I think all these behaviors are because of the cache mechanism in freshrss .
for example I have found out that when I add a feed from youtube that because I use proxy to for it and it may timeout, and it gives error, then when I try again it just gives the error fast,  I can fix this by removing the last added file in cache folder of freshrss (which usually is a zero byte file). and when I try again the freshrss seems to actually try to get the url and add it.

**Comment by rezad1393** [comment]
I right now just used the update the feeds (please rename it to update the feed with no s at the end) and the feed got the new posts.
so it seems that I am right and freshrss checks the cache file time and doesn't try the url for a minimum delay (that 15min minimum maybe?)

maybe not updating the cache file (or its time of modification) for urls that fail, is the solution?

**Comment by Alkarex** [comment]
Indeed, all feed retrieval actions are cached for a minimum of 15min, including errors, by design.
However, we should improve the case of manual actions, and we could also have different policies depending on the type of errors.
I believe we already have a couple of issues somewhere on related topics

**Comment by math-GH** [comment]
ref. #4478

**Comment by rezad1393** [comment]
> Indeed, all feed retrieval actions are cached for a minimum of 15min, including errors, by design. However, we should improve the case of manual actions, and we could also have different policies depending on the type of errors. I believe we already have a couple of issues somewhere on related topics

when I try to add a feed that has a temporary net error at the instant that I first try to add it, this behavior causes the app to not be able to add that feed for some time. this would be a bad bug for a user that doesnt know like me to go and remove the relevant cache file.

maybe a better way to handle this would be to not cache the errors at all?
I don't understand why failed tries need to be cached.
if it is for a feed that has low delay between checks, the 15min delay check still applies for feeds so that is not an issue.

**Comment by math-GH** [comment]
> maybe a better way to handle this would be to not cache the errors at all? I don't understand why failed tries need to be cached. if it is for a feed that has low delay between checks, the 15min delay check still applies for feeds so that is not an issue.

FreshRSS has a policy to consume less resources from other systems. If there is an issue with fetching an feed then it does not make sense to try to fetch the next minute again.

**Comment by Frenzie** [comment]
Though as stated this SimplePie feature seems mostly redundant with the 15 minute timeout.

**Comment by rezad1393** [comment]
is this the place for me to ask for a fix for update procedure in case of feed failure?
when I have a feed that I have set to check at every week then I have to check the freshrss to be sure that that feed didn't have an error. because if it has a temporary error at the time of checking the next check is at the next week and if the feed doesn't keep more than a week (or anything less than two weeks) I will miss posts that are not there after two weeks.

please change the update procedure in case of failures.
maybe a list of failed feeds that freshrss would check again at least one more time at after 15min delay?
